rts/Lua/LuaHandle.cpp

```diff
@@ -1100,14 +1101,16 @@ void CLuaHandle::UnitDestroyed(const CUnit* unit, const CUnit* attacker)
 	if (!cmdStr.GetGlobalFunc(L))
 		return;
 
-	static constexpr int argCount = 3 + 3;
+	static constexpr int argCount = 3 + 3 + 1;
 
 	lua_pushnumber(L, unit->id);
 	lua_pushnumber(L, unit->unitDef->id);
 	lua_pushnumber(L, unit->team);
 
 	LuaUtils::PushAttackerInfo(L, attacker);
 
+	lua_pushnumber(L, weaponDefID);
```

Comment:
There should be some sort of visibility check, like
```
if (attacker && !LuaUtils::IsUnitVisible(L, attacker))
  lua_pushnil(L);
else
  lua_pushnumber(L, weaponDefID);
```

Reply:
Looks like there's already a precedent in `UnitDamaged` that `weaponDefID` does not have to be guarded against leaking info:

https://github.com/beyond-all-reason/spring/blob/2ef0794f99d9215f44c07ec7aae57fe8c4600caa/rts/Lua/LuaHandle.cpp#L1290-L1291

I think it makes sense as even when you can't see the enemy you still can see what hit you.

Reply:
Alright.